File: Data Structures & Algorithms/duplicate-integer/submission-5.swift
class Solution {
    func hasDuplicate(_ nums: [Int]) -> Bool {
        
        /* BRUTEFORCE: double for loop check ith element w/ every other element */
        // O(n^2) time | O(1) space/memory 
        // for i in 0..<nums.count {
        //     for j in i+1..<nums.count {
        //         if nums[i] == nums[j] {
        //             return true
        //         }
        //     }
        // }
        // return false

        /* OPTIMIZED: make a set, iterate through the array and 
        check if the current element in already in the set */
        // O(n) time | O(n) space bc could be up to n elements in the Set
        var set = Set<Int>()
        
        for i in 0..<nums.count {
            if set.contains(nums[i]) {
                return true
            }
            else {
                set.insert(nums[i])
            }
        }
        return false
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-5.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        s = set()
        k = 0

        for i in range(len(nums)):
            if nums[i] not in s:
                s.add(nums[i])
                nums[k] = nums[i]
                k += 1
                
            
        return k
